query-graphs-js/src/graphPath.ts

```diff
@@ -2314,17 +2369,24 @@ function advanceWithOperation<V extends Vertex>(
     switch (currentType.kind) {
       case 'ObjectType':
         // Just take the edge corresponding to the field, if it exists and can be used.
-        const edge = nextEdgeForField(path, operation);
+        const edge = nextEdgeForField(path, operation, overrideConditions);
         if (!edge) {
           debug.groupEnd(() => `No edge for field ${field} on object type ${currentType}`);
           return { options: undefined };
         }
+        if (edge.overrideCondition) {
+          const { label } = edge.overrideCondition;
+          if (!edge.satisfiesOverrideConditions(overrideConditions)) {
+            debug.groupEnd(() => `Ignoring edge due to override label "${label}" in subgraph ${edge.head.source} on field ${field} on object type ${currentType}`);
+            return { options: undefined };
+          }
+        }
```

Comment:
This block shouldn't be needed, since satisfiability is checked inside `nextEdgeForField()`

Reply:
[2391f43](https://github.com/apollographql/federation/pull/2902/commits/2391f43bfcb9e698f38ee7da77dee9696076c82c)